avro-fastserde/src/codegen/java/com/linkedin/avro/fastserde/generated/deserialization/AVRO_1_10/FastGenericDeserializerGeneratorTest_shouldReadFixed_GenericDeserializer_3518023893123209014_3518023893123209014.java

```diff
@@ -43,31 +45,31 @@ public IndexedRecord deserializeFastGenericDeserializerGeneratorTest_shouldReadF
         } else {
             FastGenericDeserializerGeneratorTest_shouldReadFixed = new org.apache.avro.generic.GenericData.Record(readerSchema);
         }
-        byte[] testFixed0;
+        byte[] testFixed1;
         Object oldFixed0 = FastGenericDeserializerGeneratorTest_shouldReadFixed.get(0);
         if ((oldFixed0 instanceof GenericFixed)&&(((GenericFixed) oldFixed0).bytes().length == (2))) {
-            testFixed0 = ((GenericFixed) oldFixed0).bytes();
+            testFixed1 = ((GenericFixed) oldFixed0).bytes();
         } else {
-            testFixed0 = ( new byte[2]);
+            testFixed1 = ( new byte[2]);
         }
-        decoder.readFixed(testFixed0);
-        FastGenericDeserializerGeneratorTest_shouldReadFixed.put(0, new org.apache.avro.generic.GenericData.Fixed(null, testFixed0));
+        decoder.readFixed(testFixed1);
+        FastGenericDeserializerGeneratorTest_shouldReadFixed.put(0, new org.apache.avro.generic.GenericData.Fixed(testFixed0, testFixed1));
         int unionIndex0 = (decoder.readIndex());
         switch (unionIndex0) {
             case  0 :
                 decoder.readNull();
                 break;
             case  1 :
             {
-                byte[] testFixed1;
+                byte[] testFixed2;
                 Object oldFixed1 = FastGenericDeserializerGeneratorTest_shouldReadFixed.get(1);
                 if ((oldFixed1 instanceof GenericFixed)&&(((GenericFixed) oldFixed1).bytes().length == (2))) {
-                    testFixed1 = ((GenericFixed) oldFixed1).bytes();
+                    testFixed2 = ((GenericFixed) oldFixed1).bytes();
                 } else {
-                    testFixed1 = ( new byte[2]);
+                    testFixed2 = ( new byte[2]);
                 }
-                decoder.readFixed(testFixed1);
-                FastGenericDeserializerGeneratorTest_shouldReadFixed.put(1, new org.apache.avro.generic.GenericData.Fixed(null, testFixed1));
+                decoder.readFixed(testFixed2);
+                FastGenericDeserializerGeneratorTest_shouldReadFixed.put(1, new org.apache.avro.generic.GenericData.Fixed(testFixed0, testFixed2));
```

Comment:
This is unrelated to the current fix, but I think we can do slightly better than this from an object reuse point of view... we could refactor this along the lines of:

```
Object oldFixed1 = FastGenericDeserializerGeneratorTest_shouldReadFixed.get(1);
if ((oldFixed1 instanceof GenericFixed)&&(((GenericFixed) oldFixed1).bytes().length == (2))) {
    // Nothing to do
} else {
    FastGenericDeserializerGeneratorTest_shouldReadFixed.put(1, new org.apache.avro.generic.GenericData.Fixed(testFixed0, testFixed2));
}
decoder.readFixed(((GenericFixed) oldFixed1).bytes());
```

To skip one `Fixed` instance allocation.

Reply:
Sure, for object reuse perspective, we may be able to just reuse the original `Fixed` field instead of initializing one at all, right?

Reply:
Yes, that's what I mean.

Reply:
Makes sense.